lib/source/defender/aws_iot_defender_api.c

```diff
@@ -552,18 +556,39 @@ void _rejectCallback( void * pArgument,
     AwsIotDefender_Assert( pPublish->u.message.info.pPayload );
 
     /* Invoke user's callback with rejected event. */
+    _handleApplicationCallback( AWS_IOT_DEFENDER_METRICS_REJECTED, pPublish );
+    /* Delete report if exists */
+    AwsIotDefenderInternal_DeleteReport();
+}
+
+/*-----------------------------------------------------------*/
+
+void _handleApplicationCallback( AwsIotDefenderEventType_t event,
+                                 IotMqttCallbackParam_t * const pPublish )
+{
+    /* Invoke user's callback with  event. */
     AwsIotDefenderCallbackInfo_t callbackInfo;
 
     if( _startInfo.callback.function != NULL )
     {
-        callbackInfo.eventType = AWS_IOT_DEFENDER_METRICS_REJECTED;
+        callbackInfo.eventType = event;
 
         callbackInfo.pMetricsReport = AwsIotDefenderInternal_GetReportBuffer();
         callbackInfo.metricsReportLength = AwsIotDefenderInternal_GetReportBufferSize();
 
-        callbackInfo.pPayload = pPublish->u.message.info.pPayload;
-        callbackInfo.payloadLength = pPublish->u.message.info.payloadLength;
+        if( pPublish == NULL )
+        {
+            callbackInfo.pPayload = NULL;
```

Comment:
(Minor) If there is an initializer macro for the `AwsIotDefenderCallbackInfo_t` type, then the values assigned in this branch can just be set at initialization